sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -0,0 +1,285 @@
+            key=lambda x: x.times_delivered, reverse=True
+        )
+
+    def action_accept(self):
+        """Propagate recommendations to sale order."""
+        sequence = max(self.order_id.mapped("order_line.sequence") or [0])
+        order_form = Form(self.order_id.sudo())
+        to_remove = []
+        for wiz_line in self.line_ids.filtered(
+            lambda x: x.sale_line_id or x.units_included
+        ):
+            if wiz_line.sale_line_id:
+                index = self.order_id.order_line.ids.index(wiz_line.sale_line_id.id)
+                if wiz_line.units_included:
+                    with order_form.order_line.edit(index) as line_form:
+                        wiz_line._prepare_update_so_line(line_form)
+                else:
+                    to_remove.append(index)
+            else:
+                sequence += 1
+                with order_form.order_line.new() as line_form:
+                    wiz_line._prepare_new_so_line(line_form, sequence)
+        # Remove at the end and in reverse order for not having problems
+        to_remove.reverse()
+        for index in to_remove:
+            order_form.order_line.remove(index)
+        order_form.save()
+
+
+class SaleOrderRecommendationLine(models.TransientModel):
+    _name = "sale.order.recommendation.line"
+    _description = "Recommended product for current sale order"
+    _order = "id"
+
+    currency_id = fields.Many2one(related="product_id.currency_id")
+    partner_id = fields.Many2one(related="wizard_id.order_id.partner_id")
+    product_id = fields.Many2one("product.product", string="Product")
+    price_unit = fields.Monetary(compute="_compute_price_unit")
+    pricelist_id = fields.Many2one(related="wizard_id.order_id.pricelist_id")
+    times_delivered = fields.Integer(readonly=True)
+    units_delivered = fields.Float(readonly=True)
+    units_included = fields.Float()
+    wizard_id = fields.Many2one(
+        "sale.order.recommendation",
+        "Wizard",
+        ondelete="cascade",
+        required=True,
+        readonly=True,
+    )
+    sale_line_id = fields.Many2one(comodel_name="sale.order.line")
+    sale_uom_id = fields.Many2one(related="sale_line_id.product_uom")
+
+    @api.depends(
+        "partner_id",
+        "product_id",
+        "pricelist_id",
+        "units_included",
+        "wizard_id.sale_recommendation_price_origin",
+    )
+    def _compute_price_unit(self):
+        """
+        Get product price unit from product list price or from last sale price
+        """
+        price_origin = (
+            fields.first(self).wizard_id.sale_recommendation_price_origin or "pricelist"
+        )
+        for one in self:
+            if price_origin == "pricelist":
+                one.price_unit = one.product_id.with_context(
+                    partner=one.partner_id.id,
+                    pricelist=one.pricelist_id.id,
+                    quantity=one.units_included,
+                ).lst_price
```

Comment:
You are breaking the module behavior.
Is you get this field the sale order pricelist is omitted.
Please, use a method to get price from pricelist

You can take a look to unit_price in sale order lines